### PR TITLE
 feature/OB-33_RoyalPageLayout 

### PR DESCRIPTION
**Pull request by ain-odin**
1. Created a new layout for the `RoyalShaving` page in `src/Features/RoyalShaving`
2. Implemented in `RoyalShaving` 
   - some new reusable components like: `ServicePromo`, `ServiceAboutSection` and `StageHeader` from the `src/components` folder;
   - unique components `StageSection` and `StageContent`, specific only to this page, from the `src/Features/RoyalShaving` folder.
3. Edit the `Header` component from `src/components/Header` to load `li`-elements' content from API
4. Implement  the `RoyalShaving` page in the `src/app/services/royal-shaving` folder

Screenshot 2023-09-14 at 13 52 51

### CODE REVIEW

File: __tests__/page.test.tsx [hunk 1]
```diff
@@ -3,7 +3,20 @@ import { Header } from '../src/components/Header';
...
+      { title: 'Франшиза', href: '/franchising' },
+      { title: 'Инвестиции', href: '/investment' },
+      { title: 'Вакансии', href: '/vacancies' },
+      { title: 'Барберы', href: '/barbers' },
+      { title: 'Контакты', href: '/contacts' },
+      { title: 'Филиалы', href: '#' },
+    ];
+
+    render(<Header mainNavList={mainNavList} />);
     expect(screen.getByText('Главная')).toBeInTheDocument();
```
Comment (Alex-2kZharkov) [inline __tests__/page.test.tsx:21]: don't forget empty line)

File: pages/api/headerList.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    { title: 'Инвестиции', href: '/investment' },
+    { title: 'Вакансии', href: '/vacancies' },
+    { title: 'Барберы', href: '/barbers' },
+    { title: 'Контакты', href: '/contacts' },
+    { title: 'Филиалы', href: '#' },
+  ];
+
+  res.status(200).json({
+    success: true,
+    headerMainList: headerList,
```
Comment (Alex-2kZharkov) [inline pages/api/headerList.ts]: if you want to, you can rename the `headerList` to `headerMainList`, and then leave 
```
success: true,
headerMainList
```

it is called shorthand properties https://ui.dev/shorthand-properties

File: src/Features/RoyalShaving/RoyalShaving.tsx
Comment (Alex-2kZharkov) [inline src/Features/RoyalShaving/RoyalShaving.tsx:1]: please add index.ts in this folder

File: src/Features/RoyalShaving/RoyalShaving.tsx [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+import { Layout } from '../../components/Layout';
+import { Header } from '../../components/Header';
+import { LinkOptions } from '../../types/link';
+import { ServicePromo } from '../../components/ServicePromo';
+import { ServiceAboutSection } from '../../components/ServiceAboutSection';
+import { StageHeader } from './components/StageHeader/StageHeader';
+import { StageSection } from './components/StageSection';
+import { StageContent } from './components/StageContent';
+
+interface HeaderPageProps {
```
Comment (Alex-2kZharkov) [inline src/Features/RoyalShaving/RoyalShaving.tsx]: Props should have the exact same name as the component + Props postfix, so it is RoyalShavingProps

File: src/Features/RoyalShaving/components/StageContent/StageContent.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { PropsWithChildren, ReactElement } from 'react';
+
+type ContentPositionVariant = 'contentLeft' | 'contentRight';
```
Comment (Alex-2kZharkov) [inline src/Features/RoyalShaving/components/StageContent/StageContent.tsx:3]: yep, good idea

File: src/Features/RoyalShaving/components/StageHeader/StageHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+export const StageHeader = ({
+  headerTitleTop,
+  headerTitleBottom,
+  headerPosition,
+  titlePosition,
+  imagePosition,
+  children,
+}: StageHeaderProps): ReactElement => {
+  const classes = {
```
Comment (Alex-2kZharkov) [inline src/Features/RoyalShaving/components/StageHeader/StageHeader.tsx]: you can improve it a bit from the type safety perspective:

type PositionVariant = HeaderPositionVariant | TitlePositionVariant | ImagePositionVariant;

const classes: { [K in PositionVariant]: string }= {
    headerLeft: 'stage__header mr-10',
    headerRight: 'stage__heade_reversed ml-10 mr-0',
    titleLeft: 'after:-right-5',
    titleRight: 'mr-0 ml-auto after:right-auto after:-left-5 after:rotate-180',
    imageLeft: 'stage__image after:-right-5',
    imageRight: 'stage__image_reversed -left-10 after:right-auto after:left-0 after:rotate-180',
};


now, this object will cause errors if some properties are not implemented

File: src/Features/RoyalShaving/components/StageSection/StageSection.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { PropsWithChildren, ReactElement } from 'react';
+
+type StageSectionVariant = 'row' | 'rowReverse';
+
+interface StageSectionProps extends PropsWithChildren {
+  variant: StageSectionVariant;
+}
+
+export const StageSection = ({ variant, children }: StageSectionProps): ReactElement => {
+  const classes = {
```
Comment (Alex-2kZharkov) [inline src/Features/RoyalShaving/components/StageSection/StageSection.tsx]: same with type safety

File: src/app/services/royal-shaving/layout.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import { Inter } from 'next/font/google';
+import { PropsWithChildren } from 'react';
+import { ReduxProvider } from '@/store/provider';
+
+const inter = Inter({ subsets: ['latin'] });
+
+export default function RootLayout({ children }: PropsWithChildren) {
+  return (
+    <html lang="en">
+      <body className={inter.className}>
```
Comment (Alex-2kZharkov) [inline src/app/services/royal-shaving/layout.tsx:11]: @MillanaCh  I think I've missed it in your review. Please, check this out

File: src/app/services/royal-shaving/page.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  ] = await Promise.all([
+    headerListResponse.json(),
+    barbershopsInfoResponse.json(),
+    footerListsResponse.json(),
+  ]);
+
+  return (
+    <main className="overflow-hidden flex md:container md:max-w-screen-2xl mx-auto flex-col">
+      <RoyalShaving mainNavList={headerMainList} />
+      <Footer
```
Comment (Alex-2kZharkov) [inline src/app/services/royal-shaving/page.tsx:25]: the `<footer>` tag should not be inside the `<main>` tag, so you need to mve it outside

File: src/components/ServiceAboutSection/ServiceAboutSection.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export const ServiceAboutSection = ({
+  aboutSectionSrc,
+  aboutSectionAlt,
+  headline,
+  decorationVariant,
+  children,
+}: ServiceAboutSectionProps): ReactElement => {
+  const classes: Record<ServiceDecorationVariant, string> = {
```
Comment (Alex-2kZharkov) [inline src/components/ServiceAboutSection/ServiceAboutSection.tsx:20]: same

File: src/components/ServicePromo/ServicePromo.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  titleBottom: string;
+  iconVariant: ServiceIconVariant;
+}
+
+export const ServicePromo = ({
+  titleTop,
+  titleBottom,
+  iconVariant,
+}: ServicePromoProps): ReactElement => {
+  const classes: Record<ServiceIconVariant, string> = {
```
Comment (Alex-2kZharkov) [inline src/components/ServicePromo/ServicePromo.tsx:16]: see, you have it here. perfect

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]
Overall, impressive work despite the comments

File: src/Features/RoyalShaving/RoyalShaving.tsx
Comment (ain-odin) [reply src/Features/RoyalShaving/RoyalShaving.tsx:1]: That's strange. I have it there in my local folder.
 Screenshot 2023-09-15 at 11 08 58
Screenshot 2023-09-15 at 11 08 50

I'll try to recreate and push it once more

File: src/Features/RoyalShaving/RoyalShaving.tsx (hunk 3 above)
Comment (ain-odin) [reply src/Features/RoyalShaving/RoyalShaving.tsx]: My bad with inattentive copying...

File: src/components/ServiceAboutSection/ServiceAboutSection.tsx (hunk 9 above)
Comment (ain-odin) [reply src/components/ServiceAboutSection/ServiceAboutSection.tsx:20]: ?

File: src/components/ServicePromo/ServicePromo.tsx (hunk 10 above)
Comment (ain-odin) [reply src/components/ServicePromo/ServicePromo.tsx:15]: ?

File: src/Features/RoyalShaving/RoyalShaving.tsx
Comment (ain-odin) [reply src/Features/RoyalShaving/RoyalShaving.tsx:1]: I'm not sure why it isn't shown in the `Files changed` folder but `index.ts` is in local and remote branches.
local branch remote branch

File: __tests__/page.test.tsx (hunk 1 above)
Comment (ain-odin) [reply __tests__/page.test.tsx:21]: Done

File: pages/api/headerList.ts (hunk 2 above)
Comment (ain-odin) [reply pages/api/headerList.ts]: Done

File: src/Features/RoyalShaving/RoyalShaving.tsx (hunk 3 above)
Comment (ain-odin) [reply src/Features/RoyalShaving/RoyalShaving.tsx]: Done

File: src/Features/RoyalShaving/components/StageHeader/StageHeader.tsx (hunk 5 above)
Comment (ain-odin) [reply src/Features/RoyalShaving/components/StageHeader/StageHeader.tsx]: Done

File: src/app/services/royal-shaving/page.tsx (hunk 8 above)
Comment (ain-odin) [reply src/app/services/royal-shaving/page.tsx:27]: Done

**Review by Alex-2kZharkov — DISMISSED** [review]

**Review by Alex-2kZharkov — APPROVED** [review]

**Pull request merged**